core/transfer/registry/registry.go

```diff
@@ -48,6 +48,7 @@ type registryOpts struct {
 	creds         CredentialHelper
 	hostDir       string
 	defaultScheme string
+	resolver      remotes.Resolver
```

Comment:
Resolver is not serializable, what are the equivalent settings it needs to initialize it? It doesn't need to share resolvers

Reply:
Thanks for pointing this out @dmcgowan.  I can modify the implementation to ensure the necessary settings are provided directly to initialize the registry. 

I ran into issue with one setting where a pull request reporter was passed to the resolver in order to wrap the underlying http client and tracks active requests and bytes read. This is used to implement image pull progress timeout. I think I should be able to handle this using transfer.Progress.  I can give that a try but let me know if you have any suggestion.